connectors/byoc.py

```diff
@@ -204,6 +204,20 @@ async def claim(self):
         }
         await self.index.update(doc_id=self.id, doc=doc)
 
+    async def update_metadata(self, ingestion_stats=None, connector_metadata=None):
```

Comment:
Name of the method is confusing - it's updating `last_seen`, `metadata` + arbitrary set of fields passed in `ingestion_stats`.

Also what's stored inside `metadata`?

Reply:
`ingestion_stats` are common data like `indexed_document_count`.
`connector_metadata` is connector-specific data, e.g. for MySQL, it can be how many tables have been queried.

For now, it's not been implemented by any connector yet.

Reply:
Shall we drop metadata for now then?

Reply:
Hi @danajuratoni do you think we will still implement the connector-specific metadata? If so, I think we can still leave it unchanged.

Reply:
I mean, if it's not something that we're gonna do in 8.7/8.8, then why have it now? :)

Reply:
What I meant here is, we already have the `metadata` field in `.elastic-connectors-sync-jobs`, and this method accepts the optional parameter `connector_metadata`, so it can be easily ignored.

Unless @danajuratoni confirms that we are not going to implement connector-specific metadata any more, I think we can keep it for now.